recipe-client-addon/lib/RecipeRunner.jsm

```diff
@@ -50,12 +51,26 @@ this.RecipeRunner = {
       this.run();
     }
 
-    this.updateRunInterval();
-    CleanupManager.addCleanupHandler(() => timerManager.unregisterTimer(TIMER_NAME));
+    // Run once immediately after the UI is available. Do this before adding the
+    // timer so we can't end up racing it.
+    const observer = {
+      observe: (subject, topic, data) => {
+        Services.obs.removeObserver(this, UI_AVAILABLE_NOTIFICATION);
 
-    // Watch for the run interval to change, and re-register the timer with the new value
-    prefs.addObserver(RUN_INTERVAL_PREF, this);
-    CleanupManager.addCleanupHandler(() => prefs.removeObserver(RUN_INTERVAL_PREF, this));
+        this.run();
+        this.updateRunInterval();
+        CleanupManager.addCleanupHandler(() => timerManager.unregisterTimer(TIMER_NAME));
+
+        // Watch for the run interval to change, and re-register the timer with the new value
+        prefs.addObserver(RUN_INTERVAL_PREF, this);
+        CleanupManager.addCleanupHandler(() => prefs.removeObserver(RUN_INTERVAL_PREF, this));
+      },
+      QueryInterface: XPCOMUtils.generateQI([
+        Components.interfaces.nsIObserver,
+        Components.interfaces.nsISupportsWeakReference,
+      ]),
+    };
+    Services.obs.addObserver(observer, UI_AVAILABLE_NOTIFICATION);
```

Comment:
Separately, what happens if we update this add-on out of band and the notification has already happened? Would we just wait forever and never call `.run()`?

If we only want this to happen on first run, as said in the comments, I guess we want a pref of sorts.